include/gc_config_macros.h

```diff
@@ -227,10 +227,6 @@
 # define GC_CALLBACK GC_CALL
 #endif
 
-#ifndef GC_API_PATCHABLE
```

Comment:
This was never needed for mono. In il2cpp, we batch compile bdwgc into a single object (using gc.c which includes all the other.c files). This lets the compiler do more inlining, which prevents the function patching trick from working. So GC_API_PATCHABLE was needed to prevent the compiler from inlining the GC_dirty_inner function.